proc/proc_windows.go

```diff
@@ -0,0 +1,423 @@
+			return 0, nil, nil, err
+		}
+	}
+	if symtab, err = loadPETable(exe, "runtime.symtab", "runtime.esymtab"); err != nil {
+		// Same as above.
+		var err2 error
+		if symtab, err2 = loadPETable(exe, "symtab", "esymtab"); err2 != nil {
+			return 0, nil, nil, err
+		}
+	}
+	return textStart, symtab, pclntab, nil
+}
+
+func (dbp *Process) obtainGoSymbols(exe *pe.File, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	_, symdat, pclndat, err := pcln(exe)
+	if err != nil {
+		fmt.Println("could not get Go symbols", err)
+		os.Exit(1)
+	}
+
+	pcln := gosym.NewLineTable(pclndat, uint64(exe.Section(".text").Offset))
+	tab, err := gosym.NewTable(symdat, pcln)
+	if err != nil {
+		fmt.Println("could not get initialize line table", err)
+		os.Exit(1)
+	}
+
+	dbp.goSymTable = tab
+}
+
+func (dbp *Process) parseDebugLineInfo(exe *pe.File, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	if sec := exe.Section(".debug_line"); sec != nil {
+		debugLine, err := sec.Data()
+		if err != nil && uint32(len(debugLine)) < sec.Size {
+			fmt.Println("could not get .debug_line section", err)
+			os.Exit(1)
+		}
+		if 0 < sec.VirtualSize && sec.VirtualSize < sec.Size {
+			debugLine = debugLine[:sec.VirtualSize]
+		}
+		dbp.lineInfo = line.Parse(debugLine)
+	} else {
+		fmt.Println("could not find .debug_line section in binary")
+		os.Exit(1)
+	}
+}
+
+func (dbp *Process) findExecutable(path string) (*pe.File, error) {
+	if path == "" {
+		// TODO: Find executable path from PID/handle on Windows:
+		// https://msdn.microsoft.com/en-us/library/aa366789(VS.85).aspx
+		return nil, fmt.Errorf("Not yet implemented")
+	}
+	f, err := os.OpenFile(path, 0, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	peFile, err := pe.NewFile(f)
+	if err != nil {
+		return nil, err
+	}
+	data, err := peFile.DWARF()
+	if err != nil {
+		return nil, err
+	}
+	dbp.dwarf = data
+	return peFile, nil
+}
+
+func (dbp *Process) waitForDebugEvent() (threadID, exitCode int, err error) {
+	var debugEvent C.DEBUG_EVENT
+	for {
+		// Wait for a debug event...
+		res := C.WaitForDebugEvent(&debugEvent, C.INFINITE)
+		if res == C.WINBOOL(0) {
+			return 0, 0, fmt.Errorf("Could not WaitForDebugEvent")
+		}
+
+		// ... handle each event kind ...
+		unionPtr := unsafe.Pointer(&debugEvent.u[0])
+		switch debugEvent.dwDebugEventCode {
+		case C.CREATE_PROCESS_DEBUG_EVENT:
+			debugInfo := (*C.CREATE_PROCESS_DEBUG_INFO)(unionPtr)
+			dbp.os.hProcess = sys.Handle(debugInfo.hProcess)
+			_, err = dbp.addThread(sys.Handle(debugInfo.hThread), int(debugEvent.dwThreadId), false)
+			if err != nil {
+				return 0, 0, err
+			}
+			break
+		case C.CREATE_THREAD_DEBUG_EVENT:
+			debugInfo := (*C.CREATE_THREAD_DEBUG_INFO)(unionPtr)
+			_, err = dbp.addThread(sys.Handle(debugInfo.hThread), int(debugEvent.dwThreadId), false)
+			if err != nil {
+				return 0, 0, err
+			}
+			break
+		case C.LOAD_DLL_DEBUG_EVENT, C.UNLOAD_DLL_DEBUG_EVENT, C.EXIT_THREAD_DEBUG_EVENT, C.OUTPUT_DEBUG_STRING_EVENT, C.RIP_EVENT:
```

Comment:
Why leave a thread exiting unimplemented? Should be pretty simple, just remove it from the list of known threads. That way dead threads won't clutter output of `threads` command.